File: server/lib/models/work_item.dart
class WorkItem {
  WorkItem({
    required this.id,
    required this.type,
    required this.title,
    required this.description,
    required this.reproSteps,
    required this.systemInfo,
    required this.severity,
    required this.priority,
    required this.state,
    required this.areaPath,
    required this.iterationPath,
    required this.tags,
    required this.assignedTo,
    required this.createdBy,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String type;
  final String title;
  final String? description;
  final String? reproSteps;
  final String? systemInfo;
  final String severity;
  final int priority;
  final String state;
  final String areaPath;
  final String iterationPath;
  final String? tags;
  final String? assignedTo;
  final String createdBy;
  final DateTime createdAt;
  final DateTime updatedAt;

  factory WorkItem.fromRow(Map<String, Object?> r) => WorkItem(
        id: r['id']!.toString(),
        type: r['type'] as String,
        title: r['title'] as String,
        description: r['description'] as String?,
        reproSteps: r['repro_steps'] as String?,
        systemInfo: r['system_info'] as String?,
        severity: r['severity'] as String,
        priority: (r['priority'] as num).toInt(),
        state: r['state'] as String,
        areaPath: r['area_path'] as String,
        iterationPath: r['iteration_path'] as String,
        tags: r['tags'] as String?,
        assignedTo: r['assigned_to'] as String?,
        createdBy: r['created_by'] as String,
        createdAt: DateTime.parse(r['created_at']!.toString()).toUtc(),
        updatedAt: DateTime.parse(r['updated_at']!.toString()).toUtc(),
      );

  Map<String, Object?> toJson() => {
        'id': id,
        'type': type,
        'title': title,
        'description': description,
        'reproSteps': reproSteps,
        'systemInfo': systemInfo,
        'severity': severity,
        'priority': priority,
        'state': state,
        'areaPath': areaPath,
        'iterationPath': iterationPath,
        'tags': tags,
        'assignedTo': assignedTo,
        'createdBy': createdBy,
        'createdAt': createdAt.toIso8601String(),
        'updatedAt': updatedAt.toIso8601String(),
      };
}

File: server/app/models/work_item.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


def _parse_dt(value: Any) -> datetime:
    if isinstance(value, datetime):
        return value
    return datetime.fromisoformat(str(value))


class WorkItem(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str
    type: str
    title: str
    description: str | None = None
    repro_steps: str | None = Field(default=None, alias="reproSteps")
    system_info: str | None = Field(default=None, alias="systemInfo")
    severity: str
    priority: int
    state: str
    area_path: str = Field(alias="areaPath")
    iteration_path: str = Field(alias="iterationPath")
    tags: str | None = None
    assigned_to: str | None = Field(default=None, alias="assignedTo")
    created_by: str = Field(alias="createdBy")
    created_at: datetime = Field(alias="createdAt")
    updated_at: datetime = Field(alias="updatedAt")

    @classmethod
    def from_row(cls, row: dict[str, Any]) -> WorkItem:
        return cls(
            id=str(row["id"]),
            type=row["type"],
            title=row["title"],
            description=row.get("description"),
            reproSteps=row.get("repro_steps"),
            systemInfo=row.get("system_info"),
            severity=row["severity"],
            priority=int(row["priority"]),
            state=row["state"],
            areaPath=row["area_path"],
            iterationPath=row["iteration_path"],
            tags=row.get("tags"),
            assignedTo=row.get("assigned_to"),
            createdBy=row["created_by"],
            createdAt=_parse_dt(row["created_at"]),
            updatedAt=_parse_dt(row["updated_at"]),
        )

    def to_api(self) -> dict[str, Any]:
        return self.model_dump(by_alias=True, mode="json")


class WorkItemComment(BaseModel):
    id: str
    work_item_id: str = Field(alias="workItemId")
    author: str
    body: str
    created_at: datetime = Field(alias="createdAt")

    model_config = ConfigDict(populate_by_name=True)

    @classmethod
    def from_row(cls, row: dict[str, Any]) -> WorkItemComment:
        return cls(
            id=str(row["id"]),
            workItemId=str(row["work_item_id"]),
            author=row["author"],
            body=row["body"],
            createdAt=_parse_dt(row["created_at"]),
        )
